Fix Anaconda WebUI workflow so it works in upstream environment

velezd opened the pull request.

velezd committed:
Use new pty as stdin for the whole pipeline

Fixes issue with the Anaconda WebUI workflow when running Permian
without pty, eg. in GitHub runner.

pholica commented on libpermian/plugins/anaconda_webui/__init__.py:
```diff
@@ -401,7 +402,8 @@ def _start_vm(self):
 
         LOGGER.info('Running: ' + repr(cmd))
-        self.proc_virtinstall = subprocess.Popen(cmd, stdout=self.virt_install_log, stderr=subprocess.STDOUT)
+        _, pty_slave = pty.openpty()
```
I have some concerns here but it seems like only one of them will be addressable.

1. This means that there will be 2 more file descriptors opened for the whole execution of the workflow which can pose a risk of running out of file descriptors (there is a limit of 1024 file descriptors per process). The way how this could be resolved is to have the pty.openpty somehow called within the preexec_fn which is called in the new process and not in the main permian process. (I don't know now how to achieve this).
2. The file descriptors are by default inherited by the new processes unless close_fds is true and it's up to the child processes how they will handle those inherited descriptors. (this is more of a generic issue rather then related to this specific commit). https://docs.python.org/3/library/os.html#fd-inheritance
3. The file descriptors should be closed when they are not needed (after virt-install is not used).

I know I've suggested this way to resolve this issue, but at that moment, I didn't realize those issue. If it's possible to move the stdin pty handling to preexec_fn, I believe the most problematic issues will be resolved.

velezd replied:
So far I didn't find a way to utilize preexec_fn to fix this. The pty is now closed in teardown. I don't think we have to worry about running out of fds, the whole pipeline with one workflow running uses 8 and we can always raise the limit in system.

pholica commented on libpermian/cli/__init__.py:
```diff
@@ -28,6 +29,11 @@ def main(*raw_args):
     Custom CLI arguments can be optionally provided as \*args. If those are not
     provided sys.argv content is used as CLI arguments.
     """
+    # Set pseudo-terminal as fd 0 (stdin), this breaks input into pipeline but
```
I just realized it could be also worth doing this only if 0 is not a tty, like:
```python
if not os.isatty(0):
    ...
```
https://docs.python.org/3/library/os.html#os.isatty

velezd replied:
Good idea, thanks

pholica reviewed (commented):
I

rvykydal commented:
I am going to check if it fixes our issue running the test from a GH workflow on kstest runner hopefully soon.

rvykydal commented:
> I am going to check if it fixes our issue running the test from a GH workflow on kstest runner hopefully soon.

This should be the run: https://github.com/rvykydal/anaconda/actions/runs/4022656224. I copied the logs to https://rvykydal.fedorapeople.org/webui_workflow_logs/logs_19.zip as they are removed on GH after 8 days.
local_logs/*/output.txt shows:

```
2023-01-27 08:14:52 - # ----------------------------------------------------------------------^M
# testInstall (__main__.DefaultInstallation)^M
[0127/081446.748395:WARNING:discardable_shared_memory_manager.cc(197)] Less than 64MB of free space in temporary directory for shared memory files: 62^M
[0127/081446.763273:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
[0127/081446.763329:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
^M
DevTools listening on ws://127.0.0.1:9965/devtools/browser/8dbbc97c-fe9d-4551-8d4a-f94f737123c5^M
[0127/081446.767539:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.^M
[0127/081446.887472:WARNING:vaapi_wrapper.cc(834)] VAAPI video acceleration not available for angle^M
[0127/081446.889518:WARNING:sandbox_linux.cc(385)] InitializeSandbox() called with multiple threads in process gpu-process.^M
node:internal/modules/cjs/loader:827^M
      throw new ERR_UNKNOWN_BUILTIN_MODULE(request);^M
      ^^M
^M
Error [ERR_UNKNOWN_BUILTIN_MODULE]: No such built-in module: node:readline/promises^M
^[[90m    at new NodeError (node:internal/errors:387:5)^[[39m^M
^[[90m    at Function.Module._load (node:internal/modules/cjs/loader:827:13)^[[39m^M
^[[90m    at Module.require (node:internal/modules/cjs/loader:1057:19)^[[39m^M
^[[90m    at require (node:internal/modules/cjs/helpers:103:18)^[[39m^M
    at Object.<anonymous> ^[[90m(/root/workdir/anaconda/^[[39mui/webui/test/common/chromium-cdp-driver.js:31:18^[[90m)^[[39m^M
^[[90m    at Module._compile (node:internal/modules/cjs/loader:1155:14)^[[39m^M
^[[90m    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)^[[39m^M
^[[90m    at Module.load (node:internal/modules/cjs/loader:1033:32)^[[39m^M
^[[90m    at Function.Module._load (node:internal/modules/cjs/loader:868:12)^[[39m^M
^[[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)^[[39m {^M
  code: ^[[32m'ERR_UNKNOWN_BUILTIN_MODULE'^[[39m^M
}^M
Traceback (most recent call last):^M
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod^M
    method()^M
  File "/root/workdir/anaconda/./ui/webui/test/integration/default.py", line 33, in testInstall^M
    self._testInstall()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 97, in _testInstall^M
    self.installer.open()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/installer.py", line 83, in open^M
    self.browser.open(f"/cockpit/@localhost/anaconda-webui/index.html#/{step}")^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 232, in open^M
    self._set_window_size(size[0], size[1])^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 971, in _set_window_size^M
    self.cdp.invoke("Emulation.setDeviceMetricsOverride",^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 215, in command^M
    self.start()^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 328, in start^M
    self.invoke("Page.addScriptToEvaluateOnNewDocument", source=src, no_trace=True)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 222, in command^M
    raise RuntimeError("CDP broken")^M
RuntimeError: CDP broken^M
Journal extracted to DefaultInstallation-testInstall-fedora-37-192.168.122.70-22-FAIL.log.gz^M
rsync: [sender] link_stat "/tmp/anaconda-tb-*" failed: No such file or directory (2)^M
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1823) [Receiver=3.2.3]^M
rsync: [Receiver] write error: Broken pipe (32)^M
Traceback (most recent call last):^M
  File "/root/workdir/anaconda/./ui/webui/test/integration/default.py", line 33, in testInstall^M
    self._testInstall()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 97, in _testInstall^M
    self.installer.open()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/installer.py", line 83, in open^M
    self.browser.open(f"/cockpit/@localhost/anaconda-webui/index.html#/{step}")^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 232, in open^M
    self._set_window_size(size[0], size[1])^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 971, in _set_window_size^M
    self.cdp.invoke("Emulation.setDeviceMetricsOverride",^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 215, in command^M
    self.start()^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 328, in start^M
    self.invoke("Page.addScriptToEvaluateOnNewDocument", source=src, no_trace=True)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 222, in command^M
    raise RuntimeError("CDP broken")^M
RuntimeError: CDP broken^M
^M
# Result testInstall (__main__.DefaultInstallation) failed^M
# 1 TEST FAILED [6s on d44e4e8dc475]^M
```

The issue does not seem to be ralated to the pty patch, if I revert it, I am getting it as well:
https://rvykydal.fedorapeople.org/webui_workflow_logs/logs_20.zip 
```
2023-01-27 08:57:14 - time="2023-01-27T08:57:07Z" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
# ----------------------------------------------------------------------^M
# testInstall (__main__.DefaultInstallation)^M
[0127/085709.242036:WARNING:discardable_shared_memory_manager.cc(197)] Less than 64MB of free space in temporary directory for shared memory files: 62^M
[0127/085709.257475:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
[0127/085709.257544:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
^M
DevTools listening on ws://127.0.0.1:9382/devtools/browser/d2bf7db5-b163-4031-92f3-da69a7ec0bdb^M
[0127/085709.261590:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.^M
[0127/085709.380283:WARNING:vaapi_wrapper.cc(834)] VAAPI video acceleration not available for angle^M
[0127/085709.382224:WARNING:sandbox_linux.cc(385)] InitializeSandbox() called with multiple threads in process gpu-process.^M
node:internal/modules/cjs/loader:827^M
      throw new ERR_UNKNOWN_BUILTIN_MODULE(request);^M
      ^^M
^M
Error [ERR_UNKNOWN_BUILTIN_MODULE]: No such built-in module: node:readline/promises^M
^[[90m    at new NodeError (node:internal/errors:387:5)^[[39m^M
^[[90m    at Function.Module._load (node:internal/modules/cjs/loader:827:13)^[[39m^M
^[[90m    at Module.require (node:internal/modules/cjs/loader:1057:19)^[[39m^M
^[[90m    at require (node:internal/modules/cjs/helpers:103:18)^[[39m^M
    at Object.<anonymous> ^[[90m(/root/workdir/anaconda/^[[39mui/webui/test/common/chromium-cdp-driver.js:31:18^[[90m)^[[39m^M
^[[90m    at Module._compile (node:internal/modules/cjs/loader:1155:14)^[[39m^M
^[[90m    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)^[[39m^M
^[[90m    at Module.load (node:internal/modules/cjs/loader:1033:32)^[[39m^M
^[[90m    at Function.Module._load (node:internal/modules/cjs/loader:868:12)^[[39m^M
^[[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)^[[39m {^M
  code: ^[[32m'ERR_UNKNOWN_BUILTIN_MODULE'^[[39m^M
}^M
Traceback (most recent call last):^M
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod^M
    method()^M
  File "/root/workdir/anaconda/./ui/webui/test/integration/default.py", line 33, in testInstall^M
    self._testInstall()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 97, in _testInstall^M
    self.installer.open()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/installer.py", line 83, in open^M
    self.browser.open(f"/cockpit/@localhost/anaconda-webui/index.html#/{step}")^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 232, in open^M
    self._set_window_size(size[0], size[1])^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 971, in _set_window_size^M
    self.cdp.invoke("Emulation.setDeviceMetricsOverride",^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 215, in command^M
    self.start()^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 328, in start^M
    self.invoke("Page.addScriptToEvaluateOnNewDocument", source=src, no_trace=True)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 222, in command^M
    raise RuntimeError("CDP broken")^M
RuntimeError: CDP broken^M
Journal extracted to DefaultInstallation-testInstall-fedora-37-192.168.122.226-22-FAIL.log.gz^M
rsync: [sender] link_stat "/tmp/anaconda-tb-*" failed: No such file or directory (2)^M
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1823) [Receiver=3.2.3]^M
rsync: [Receiver] write error: Broken pipe (32)^M
Traceback (most recent call last):^M
  File "/root/workdir/anaconda/./ui/webui/test/integration/default.py", line 33, in testInstall^M
    self._testInstall()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 97, in _testInstall^M
    self.installer.open()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/installer.py", line 83, in open^M
    self.browser.open(f"/cockpit/@localhost/anaconda-webui/index.html#/{step}")^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 232, in open^M
    self._set_window_size(size[0], size[1])^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 971, in _set_window_size^M
    self.cdp.invoke("Emulation.setDeviceMetricsOverride",^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 215, in command^M
    self.start()^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 328, in start^M
    self.invoke("Page.addScriptToEvaluateOnNewDocument", source=src, no_trace=True)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 205, in invoke^M
    res = self.command(cmd)^M
  File "/root/workdir/anaconda/ui/webui/test/common/cdp.py", line 222, in command^M
    raise RuntimeError("CDP broken")^M
RuntimeError: CDP broken^M
^M
# Result testInstall (__main__.DefaultInstallation) failed^M
# 1 TEST FAILED [6s on 2eddf50bf9bf]^M
```

And these are the logs we were looking at when you discovered the pty issue: https://rvykydal.fedorapeople.org/webui_workflow_logs/logs_15.zip 
```
2023-01-16 15:13:07 - time="2023-01-16T14:36:52Z" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
# ----------------------------------------------------------------------^M
# testInstall (__main__.DefaultInstallation)^M
[0116/143654.441527:WARNING:discardable_shared_memory_manager.cc(197)] Less than 64MB of free space in temporary directory for shared memory files: 62^M
[0116/143654.456891:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
[0116/143654.456957:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
^M
DevTools listening on ws://127.0.0.1:9267/devtools/browser/85a5102a-796f-416c-92b9-0805619e3b1c^M
[0116/143654.462777:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.^M
[0116/143654.495724:WARNING:sandbox_linux.cc(380)] InitializeSandbox() called with multiple threads in process gpu-process.^M
[0116/143654.888444:ERROR:devtools_http_handler.cc(636)] Using unsafe HTTP verb GET to invoke /json/new. This action will stop supporting GET and POST verbs in future versions.^M
> info: conf:  ^M
> info: ^M
> log: Selecting one or less disks by default: sda^M
rsync: [sender] link_stat "/tmp/anaconda-tb-*" failed: No such file or directory (2)^M
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1823) [Receiver=3.2.3]^M
rsync: [Receiver] write error: Broken pipe (32)^M
> log: Killing WebUI process, PID:  2230^M
> error: Localization client closed^M
> error: Storage client closed^M
> error: Payloads client closed^M
> error: Boss client closed^M
client_loop: send disconnect: Broken pipe^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.107 port 22: Connection refused^M^M
Traceback (most recent call last):^M
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod^M
    method()^M
  File "/root/workdir/anaconda/./ui/webui/test/integration/default.py", line 33, in testInstall^M
    self._testInstall()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 105, in _testInstall^M
    self.reboot()^M
  File "/root/workdir/anaconda/ui/webui/test/helpers/integration.py", line 91, in reboot^M
    self.machine.wait_reboot()^M
  File "/root/workdir/anaconda/ui/webui/bots/machine/machine_core/ssh_connection.py", line 149, in wait_reboot^M
    raise exceptions.Failure("Timeout waiting for system to reboot properly")^M
machine_core.exceptions.Failure: Timeout waiting for system to reboot properly^M
Wrote screenshot to DefaultInstallation-testInstall-fedora-37-192.168.122.107-22-FAIL.png^M
Wrote HTML dump to DefaultInstallation-testInstall-fedora-37-192.168.122.107-22-FAIL.html^M
Wrote JS log to DefaultInstallation-testInstall-fedora-37-192.168.122.107-22-FAIL.js.log^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
ssh: connect to host 192.168.122.107 port 22: Connection timed out^M^M
Unexpected exception while running testInstall (__main__.DefaultInstallation)^M
Traceback (most recent call last):^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 2310, in runOne^M
    unittest.TestSuite([test]).run(result)^M
  File "/usr/lib64/python3.9/unittest/suite.py", line 122, in run^M
    test(result)^M
  File "/usr/lib64/python3.9/unittest/case.py", line 651, in __call__^M
    return self.run(*args, **kwds)^M
  File "/usr/lib64/python3.9/unittest/case.py", line 594, in run^M
    with outcome.testPartExecutor(self):^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 1273, in testPartExecutor^M
    failureHandler(exc_info, False)^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 1249, in failureHandler^M
    test_case.copy_journal("FAIL")^M
  File "/root/workdir/anaconda/ui/webui/test/common/testlib.py", line 2016, in copy_journal^M
    m.execute("journalctl|gzip", stdout=fp)^M
  File "/root/workdir/anaconda/ui/webui/bots/machine/machine_core/ssh_connection.py", line 277, in execute^M
    self._ensure_ssh_master()^M
  File "/root/workdir/anaconda/ui/webui/bots/machine/machine_core/ssh_connection.py", line 249, in _ensure_ssh_master^M
    self._start_ssh_master()^M
  File "/root/workdir/anaconda/ui/webui/bots/machine/machine_core/ssh_connection.py", line 202, in _start_ssh_master^M
    raise exceptions.Failure("SSH master process exited with code: {0}".format(proc.returncode))^M
machine_core.exceptions.Failure: SSH master process exited with code: 255^M
# 1 TEST FAILED [2174s on c1870877515d]^M

```

velezd committed:
Modify Anaconda WebUI worflow Containerfile to use Fedora

Cockpit test framework now requires newer node.js then what was
available in Centos Stream 9. Also Cockpit team is using Fedora
for their testing so this should prevent similar issues in the
future.

rvykydal commented:
The run on kstest-runner (https://github.com/rvykydal/anaconda/actions/runs/4123963266) seems to succeed, output.txt from the logs artifact from the run:
```
2023-02-08 12:26:00 - # ----------------------------------------------------------------------^M
# testInstall (__main__.DefaultInstallation.testInstall)^M
[0208/121642.456638:WARNING:discardable_shared_memory_manager.cc(197)] Less than 64MB of free space in temporary directory for shared memory files: 62^M
[0208/121642.471131:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
[0208/121642.471256:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory^M
^M
DevTools listening on ws://127.0.0.1:10163/devtools/browser/c4854b53-057f-4f9d-bb7a-19eb87263470^M
[0208/121642.474915:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.^M
[0208/121642.564434:WARNING:vaapi_wrapper.cc(834)] VAAPI video acceleration not available for angle^M
[0208/121642.566146:WARNING:sandbox_linux.cc(385)] InitializeSandbox() called with multiple threads in process gpu-process.^M
[0208/121642.900147:ERROR:devtools_http_handler.cc(645)] Using unsafe HTTP verb GET to invoke /json/new. This action will stop supporting GET and POST verbs in future versions.^M
> info: conf:  ^M
> info: conf:  ^M
> info: ^M
> log: Selecting one or less disks by default: sda^M
rsync: [sender] link_stat "/tmp/anaconda-tb-*" failed: No such file or directory (2)^M
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1861) [Receiver=3.2.7]^M
rsync: [Receiver] write error: Broken pipe (32)^M
> log: Killing WebUI process, PID:  2224^M
> error: Localization client closed^M
> error: Localization client closed^M
> error: Storage client closed^M
> error: Storage client closed^M
> error: Payloads client closed^M
> error: Payloads client closed^M
> error: Boss client closed^M
> error: Boss client closed^M
client_loop: send disconnect: Broken pipe^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
ssh: connect to host 192.168.122.169 port 22: Connection refused^M^M
# Result testInstall (__main__.DefaultInstallation.testInstall) succeeded^M
# 1 TEST PASSED [558s on 44ecdaa9b521]^M
```

rvykydal reviewed (approved):
Looks good to me, thank you.
